tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,300 @@
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_BUF_FREE_BY_NNSTREAMER			= 0, /**< Default. Application should not deallocate this buffer. NNStreamer will deallocate when the buffer is no more needed */
+  NNS_BUF_NO_NOT_FREE				= 1, /**< This buffer is not to be freed. */
+} nns_buf_policy;
+
+/**
+ * @brief Enumeration for nns pipeline state
+ * @detail Refer to https://gstreamer.freedesktop.org/documentation/design/states.html
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI) Fill with GST's state values.
+ * @todo Check the consistency against MMFW APIs.
+ */
+typedef enum {
+  NNS_PIPELINE_UNKNOWN				= 0, /**< Unknown state. Maybe not constructed? */
+  NNS_PIPELINE_NULL,				= 1, /**< GST-State "Null" */
+  NNS_PIPELINE_READY,				= 2, /**< GST-State "Ready" */
+  NNS_PIPELINE_PAUSED,				= 3, /**< GST-State "Paused" */
+  NNS_PIPELINE_PLAYING,				= 4, /**< GST-State "Playing" */
+} nns_pipeline_state;
+
+/**
+ * @brief Enumeration for switch types
+ * @detail This designates different GStreamer filters, "GstInputSelector"/"GetOutputSelector".
+ * @since_tizen 5.5
+ * @todo There may be more filters that can be supported.
+ */
+typedef enum {
+  NNS_SWITCH_OUTPUTSELECTOR,			= 0, /** GstOutputSelector */
+  NNS_SWITCH_INPUTSELECTOR,			= 1, /** GstInputSelector */
+} nns_switch_type;
+
+
+/****************************************************
+ ** NNStreamer Pipeline Construction (gst-parse)   **
+ ****************************************************/
+/**
+ * @brief Construct the pipeline (GStreamer + NNStreamer)
+ * @details Use this function to create a gst_parse_launch compatible NNStreamer pipelines.
+ * @since_tizen 5.5
+ * @param[in] pipeline_description The pipeline description compatible with GStreamer gst_parse_launch(). Refer to GStreamer manual or NNStreamer (github.com/nnsuite/nnstreamer) documentation for examples and the grammar.
+ * @param[out] pipe The nnstreamer pipeline handler from the given description
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_construct (const char *pipeline_description, nns_pipeline_h *pipe);
+
+/**
+ * @brief Destroy the pipeline
+ * @details Use this function to destroy the pipeline constructed with nns_construct_pipeline.
+ * @since_tizen 5.5
+ * @param[in] pipe The pipeline to be destroyed.
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_destroy (nns_pipeline_h *pipe);
+
+/**
+ * @brief Get the state of pipeline
+ * @detail Get the state of The pipeline handle returned by nns_construct_pipeline (pipe).
+ * @since_tizen 5.5
+ * @param[in] pipe The pipeline to be monitored.
+ * @param[out] state The pipeline state.
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_getstate (nns_pipeline_h *pipe, nns_pipeline_state *state);
```

Comment:
We need another function to set the pipeline state - `int nns_pipeline_setstate(nns_pipeline_h *pipe, nns_pipeline_state *state)`.

Reply:
Yes. I'd want it abstracted with start/stop(). I was thinking about using stop as pause and start as resume as well; a pipeline "permanently stops" with destructor.

Reply:
start/stop() will abstract to just two states, however, we might need to provide interface with states - 
https://gstreamer.freedesktop.org/documentation/application-development/advanced/pipeline-manipulation.html#play-a-section-of-a-media-file
For example, this example needs interface to PAUSED states.

We can start with this interface and add access to more states based on application requirement.

Reply:
Refer to Line 93-109. I've reduced the number of functions to simplify usage for app developers.

Reply:
Works